1 Niagara 4 Overview Exercise - BFile/myFile-rt/src/com/tridiumuniversity/myFile/BFile.java

```diff
@@ -0,0 +1,213 @@
+   * @see #path
+   */
+  public void setPath(String v) { setString(path, v, null); }
+
+  //endregion Property "path"
+
+  //region Property "size"
+
+  /**
+   * Slot for the {@code size} property.
+   * @see #getSize
+   * @see #setSize
+   */
+  public static final Property size = newProperty(0, 0, null);
+
+  /**
+   * Get the {@code size} property.
+   * @see #size
+   */
+  public int getSize() { return getInt(size); }
+
+  /**
+   * Set the {@code size} property.
+   * @see #size
+   */
+  public void setSize(int v) { setInt(size, v, null); }
+
+  //endregion Property "size"
+
+  //region Property "createdDate"
+
+  /**
+   * Slot for the {@code createdDate} property.
+   * @see #getCreatedDate
+   * @see #setCreatedDate
+   */
+  public static final Property createdDate = newProperty(0, BAbsTime.now(), null);
+
+  /**
+   * Get the {@code createdDate} property.
+   * @see #createdDate
+   */
+  public BAbsTime getCreatedDate() { return (BAbsTime)get(createdDate); }
+
+  /**
+   * Set the {@code createdDate} property.
+   * @see #createdDate
+   */
+  public void setCreatedDate(BAbsTime v) { set(createdDate, v, null); }
+
+  //endregion Property "createdDate"
+
+  //region Property "modifiedDate"
+
+  /**
+   * Slot for the {@code modifiedDate} property.
+   * @see #getModifiedDate
+   * @see #setModifiedDate
+   */
+  public static final Property modifiedDate = newProperty(0, BAbsTime.now(), null);
+
+  /**
+   * Get the {@code modifiedDate} property.
+   * @see #modifiedDate
+   */
+  public BAbsTime getModifiedDate() { return (BAbsTime)get(modifiedDate); }
+
+  /**
+   * Set the {@code modifiedDate} property.
+   * @see #modifiedDate
+   */
+  public void setModifiedDate(BAbsTime v) { set(modifiedDate, v, null); }
+
+  //endregion Property "modifiedDate"
+
+  //region Action "touch"
+
+  /**
+   * Slot for the {@code touch} action.
+   * @see #touch()
+   */
+  public static final Action touch = newAction(Flags.ASYNC, null);
+
+  /**
+   * Invoke the {@code touch} action.
+   * @see #touch
+   */
+  public void touch() { invoke(touch, null, null); }
+
+  //endregion Action "touch"
+
+  //region Action "print"
+
+  /**
+   * Slot for the {@code print} action.
+   * @see #print()
+   */
+  public static final Action print = newAction(Flags.ASYNC, null);
+
+  /**
+   * Invoke the {@code print} action.
+   * @see #print
+   */
+  public void print() { invoke(print, null, null); }
+
+  //endregion Action "print"
+
+  //region Type
+
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BFile.class);
+
+  //endregion Type
+
+//@formatter:on
+//endregion /*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+    public void doTouch(){
+        setModifiedDate(BAbsTime.now());
+    }
+    public void doPrint(){
+        System.out.printf("Filename = %s, Path = %s, Size = %s, Created Date = %s, Modified Date %s%n",getFileName(),getPath(),getSize(),getCreatedDate(),getModifiedDate());
```

Comment:
This is fine, but if you want the output to be more localized, you could look into accepting a `Context` parameter in your `doPrint` method and then doing something like `getCreatedDate().toString(cx)` and `getModifiedDate().toString(cx)`.

Reply:
And to clarify, this would result in it displaying the user's timezone rather than the JACEs timezone?  Thanks for the feedback.

Reply:
Well, I thought it would, but I just did some tests and it looks like although the time is capable of formatting itself based on information like the timezone if it's included in the `Context`, I don't see where this information is being included in the `Context` for you by default. I don't have a JACE handy in another timezone to test with, so maybe the framework would include the user's timezone in the `Context` if it detected that it was different than the JACE's. But I would do some testing before relying on that functionality.